src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs

```diff
@@ -389,6 +389,12 @@ public void InstallService(string serviceName, string serviceDisplayName, string
                     throw new InvalidOperationException(StringUtil.Loc("OperationFailed", nameof(CreateService), GetLastError()));
                 }
 
+                //invoke the service with special argument, that tells it to register an event log trace source (need to run as an admin)
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
```

Comment:
This is fine for now. Per offline conversation, it sounds like pinvoke isn't a clean option. I wonder if powershell or cmd.exe would be easier.